### PR TITLE
Rename app and drop e2etest

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward giantswarm/giantswarm#6078

Rename its chart to `*-app`, and drop whole e2e test. 

For more information, please see our App catalog phase 2 spec documentation. 
https://github.com/giantswarm/giantswarm/blob/master/specs/app-catalog-phase-2.md#helm-charts--git-repositories

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,154 +2,16 @@ version: 2.1
 orbs:
   architect: giantswarm/architect@0.2.0
 
-e2eTest: &e2eTest
```
Comment (tomahawk28) [inline .circleci/config.yml]: Drop whole e2e test from the same repo

Comment (rossf7) [reply .circleci/config.yml]: Yes, the managedservices test only works with appr. We will add a new e2e test called basicapp that is compatible with App Catalog. We can re-enable this then.

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,154 +2,16 @@ version: 2.1
...
     jobs:
-      - build
-
       - architect/push-to-app-catalog:
-          name: "package and push kubernetes-metrics-server chart"
+          name: "package and push metrics-server-app chart"
           app_catalog: "giantswarm-catalog"
           app_catalog_test: "giantswarm-test-catalog"
-          chart: "kubernetes-metrics-server-chart"
+          chart: "metrics-server-app"
```
Comment (rossf7) [inline .circleci/config.yml]: Sorry we need to use `default-catalog` and `default-test-catalog` here. See https://github.com/giantswarm/chart-operator/pull/286#issue-317932168.

File: helm/metrics-server-app/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
 appVersion: 0.3.3
 description: Metrics Server is a cluster-wide aggregator of resource usage data.
-name: kubernetes-metrics-server-chart
+name: metrics-server-app
```
Comment (rossf7) [inline helm/metrics-server-app/Chart.yaml]: Let's bump the version below to 0.4.0 and add a changelog entry. See https://github.com/giantswarm/kube-state-metrics-app/blob/master/CHANGELOG.md.

File: .circleci/config.yml (hunk 2 above)
Comment (tomahawk28) [reply .circleci/config.yml]: I already follow as you did, Thanks anyway.

**Pull request merged**